src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,54 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private int size;
+    private Object[] keys;
+    private Object[] values;
+
+    public StorageImpl() {
+        this.keys = new Object[MAX_SIZE];
+        this.values = new Object[MAX_SIZE];
+        this.size = 0;
+    }
+
+    private int getIndex(K key) {
+        for (int i = 0; i < size; i++) {
+            if (key == null && keys[i] == null) {
+                return i;
+            } else if (keys[i] != null && keys[i].equals(key)) {
+                return i;
+            }
```

Comment:
when you have same body in if and else then it for sure may be combined
```suggestion
            if (key == keys[i] || key != null && key.equals(keys[i])) {
                return i;
            }
```